raiden/network/pathfinding.py

```diff
@@ -157,6 +158,42 @@ def get_pfs_info(url: str) -> Optional[PFSInfo]:
         return None
 
 
+def get_valid_pfs_url(
+    service_registry: ServiceRegistry,
+    index_in_service_registry: int,
+    block_identifier: BlockSpecification,
+) -> Optional[str]:
+    """Returns the URL for the PFS identified by the given index
+
+    Checks validity of registration in the ServiceRegistry contract and
+    checks the price from the PFS' info endpoint.
+
+    Returns PFS URL or None (if any check fails).
+    """
+    address = service_registry.ever_made_deposits(
+        block_identifier=block_identifier, index=index_in_service_registry
+    )
+    if not address:
```

Comment:
Can this ever happen?

Reply:
If we're using `block_identifier == "latest"`, I think it can, although unlikely.